### ISSUE TITLE
Cypress side panel seemingly disappears on redirect

### ISSUE THREAD

**Issue by WesselKuipers** [cypress-io/cypress#18958]
### Current behavior

When testing against the staging environment of my application using `cypress open`, subsequent visits to the app seems to cause the side bar to disappear and halt test execution. 

After removing the files from `File` -> `View App Files`, the first test run will run like before and any subsequent run will once again cause the side menu to disappear.

This appears to be the case in both Windows and on Linux as well as across multiple browsers.

As far as I can tell this seems to be specific to the application, but I haven't been able to pinpoint exactly what is going wrong here. It also doesn't occur when running in headless mode using `cypress run`.

In the screencast below I first clear the files, run a test, close the browser, and attempt to run it again.

https://user-images.githubusercontent.com/4731930/142214608-0efeb3be-b6eb-4e38-8766-fb7248368822.mp4

### Desired behavior

The Cypress test runner environment should persist or give more information about what's possibly causing issues when testing.

### Test code to reproduce

Fork for the test repo:
https://github.com/WesselKuipers/cypress-test-tiny

### Cypress Version

9.0.0

**Comment by BlueWinds** [comment]
It looks like appsemble is using service workers, which may be the cause.

The way cypress works is by registering itself for the domain name under test, and proxying all calls through our server. This tends to cause issues with service workers that try to cache or otherwise process requests - they're not aware of the cypress magic going on under the hood, and can end up breaking cypress in interesting ways.

Is it possible to disable your service worker, to see if this is indeed the problem?

**Comment by remcohaszing** [comment]
I’m a coworker of @WesselKuipers 

I created a merge request in [appsemble/appsemble!2250](https://gitlab.com/appsemble/appsemble/-/merge_requests/2250) to disable service workers. The relevant app is available on https://empty.appsemble.2250.appsemble.review

I have confirmed the issue goes away when chaning the URL from https://empty.appsemble.2250.appsemble.review to https://empty.appsemble.2250.appsemble.review.

Removing the service worker caused an uncaught promise rejection which can be ignored by adding the following snippet:

```js
Cypress.on('uncaught:exception', (err, runnable, promise) => {
  if (promise) {
    return false
  }
})
```

We can keep this review environment up if it helps.
